### ISSUE TITLE
Routing Behavior around Corners is incorrect

### ISSUE THREAD

**Issue by ChristianHirt** [JuPedSim/jpscore#1056]
**JuPedSim - JPScore**

- Current date   : 02/02/2023
- Version        : 
- Compiler       :
- Commit hash    :
- Commit date    :
- Branch         : SUMO-Integration
- Python         :

**Describe the bug**
transferred Demo file 3 is not simulating pedestrians walking around a Corner as expected. Simulations do not end properly with some journeys. In detail:
When executing a transferred version of Demo file 3 with one agent and  one way point at the exit Area the agent runs into a wall and the simulation never ends. _Can be seen in file `03_1_agent_one_way_point`_.
Another Version with five agents and one way point leads to the simulation crashing with an `ZeroDivisionError: float division by zero` after 1230 Iteration. _Can be seen in file `03_5_agents_one_way_point`_.
When executing a version with additional way points the simulations is working correctly. _Can be seen in file `03_working`_.
However when changing the way point (16, -0.5) 's y value to a number below -0.5 the agent will run into the wall and gets stuck resulting in the simulation never ending. _Can be seen in file `03_not_working`_. The only difference in both files is the way point changed to  (16, -0.5001)

**Expected behavior**
No matter how many or how way points are placed the simulation should not get stuck or crash 

**Screenshots**
all executed simulations and their output is found [here](https://github.com/JuPedSim/jpscore/files/10570212/incorrect_simulations.zip) or at the bottom in zip file.

**Additional context**
the trajectories are created without the comment including the geometry path. i added this line in all trajectories in order to be displayed better in jpsvis. All simulations the do not end have been interrupted using `^C`. 

[incorrect_simulations.zip](https://github.com/JuPedSim/jpscore/files/10570212/incorrect_simulations.zip)

**Comment by ChristianHirt** [comment]
Upon further trial and error, I noticed other problems of a similar nature.
First, I changed the working file so that the way point in the corner is further to the right. this results in an Zerodevisionerror but also shows that something with the routing does not seem right. All the Agents are hugging the wall trying to get further upwards rather that to the right. _Can be seen in file `w2_further_back`_.
This strange behavior also happens when the the radius of the first waypoint is smaller. The first way point lies inside the bottleneck. _Can be seen in file `w1_adapted`_.
It also happend when the radius of the second way point was made smaller. _Can be seen in file `w2_radius`_.

All futher examples can be seen [here](https://github.com/JuPedSim/jpscore/files/10706266/more_examples.zip)
